File: HelloWorld/procedural_1.rb
#!/usr/bin/env ruby
# encoding: utf-8

# Array.size() is not procedural: it's a method of the object Array.
if ARGV.size == 0 then
  name = 'World'
else
  name = ARGV[ 0 ]
end
puts "Hello, #{ name }!"

puts '» José Carlos Monteiro «'

File: HelloWorld/Procedural1.pas
program Procedural1;

uses
  SysUtils;

var
  name : string = 'World';

begin
  if ParamCount = 1 then
    name := ParamStr( 1 );

  WriteLn( Format( 'Hello, %s!', [ name ] ) );

  WriteLn( '» José Carlos Monteiro «' );
end.
